Assets/ScreenShake/ScreenShake.cs

```diff
@@ -0,0 +1,57 @@
+﻿/*
+ * Copyright © 2015 Rustic 
+ * This file is part of the Rustic Unity Toolset.
+ *
+ * The Rustic Unity Toolset is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * The Rustic Unity Toolset is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with The Rustic Unity Toolset.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+using UnityEngine;
+using System.Collections;
+
+namespace Rustic {
+  public class ScreenShake : MonoBehaviour {
```

Comment:
An alternate implementation that jumps to mind is:
Represent the screen shake behavior as just a static method with a signature like `IEnumerator Shake(camera, x, y, duration, frequency)`, and let the component that wants to invoke that behavior (in this case, the button) call that and then call `StartCoroutine` on the result.

My gut says that this approach is functionally equivalent but conceptually simpler, and probably slightly better performance-wise since having a whole MonoBehavior to invoke the shaking involves creating and disposing of a gameobject and a transform.

I'm not asking you to change it or anything, I mainly just want to have you double-check my information and my logic, and to let me know if I'm overlooking some benefit of the current approach or some detriment to my alternative.